qualtop/wizard/account_invoice_refund.py

```diff
@@ -0,0 +1,26 @@
+from odoo import models, api
+
+
+class AccountInvoiceRefund(models.TransientModel):
+    _inherit = "account.invoice.refund"
+
+    @api.multi
+    def invoice_refund(self):
+        res = super(AccountInvoiceRefund, self).invoice_refund()
+        data_refund = self.read(['filter_refund'])[0]['filter_refund']
+        if data_refund == 'refund' and res.get('domain', False):
+            """ To obtain the credit note id is necessary search in
+            res['domain'] which has the form
+            [('type', '=', 'out_refund'), ('id', 'in', [5])]. The credit note
+            id is in res['domain'][1][2][0]
+            """
+            cn_id = res['domain'][1][2][0]
```

Comment:
If the domain is `[('type', '=', 'out_refund'), ('id', 'in', [5])]` you should search the `id` key, instead of process the domain with hard-coded indexes.

Reply:
What about if the `invoice_refund` is called for 2 or more invoices?

Reply:
1 - The credit note id is only in ```res['domain']```

2 - The method ```invoice_refund``` is called from wizard in a specific invoice, no mora that one invoice.

Reply:
That is not a good `why`
Since that you can install other modules calling all kind of methods with multi.

But I'm reviewing the main method and this one has the same behavior.
Then +1 with this piece

Reply:
To get the id why is not a good option for you search the `id`
In the future you will have errors if the order changes

We have a similar issue related with l10n_mx

Reply:
I mean,
```
for item in domain:
 if len(item) == 3 and item[0] == 'id':
```

Reply:
What do you this comment?
https://github.com/teamsye/qualtop/pull/6#discussion_r208361192